server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/template_configs/templates.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define(['mithril', 'lodash', 'string-plus', 'helpers/mrequest', 'js-routes', 'models/model_mixins'],
+  function (m, _, s, mrequest, Routes, Mixins) {
+    var Templates = function (data) {
+      Mixins.HasMany.call(this, {
+        factory:    Templates.Template.fromJSON,
+        as:         'Template',
+        collection: data,
+        uniqueOn:   'name'
+      });
+    };
+
+    Templates.Template = function (data) {
+      this.parent    = Mixins.GetterSetter();
+      this.name      = m.prop(s.defaultToIfBlank(data.name, ''));
+      this.url       = m.prop(s.defaultToIfBlank(data.url, ''));
+      this.pipelines = m.prop(_.map(data.pipelines, function (pipeline) {
+        return new function () {
+          this.name = m.prop(pipeline.name);
+          this.url  = m.prop(pipeline._links.self.href);
+        };
+      }));
+
+      this.delete = function () {
+        return m.request({
+          method: 'DELETE',
+          url:    Routes.apiv3AdminTemplatePath({template_name: data.name}),
+          config: mrequest.xhrConfig.v4
+        });
+      };
+    };
+
+    Templates.Template.fromJSON = function (data) {
+      return new Templates.Template({
+        name:      data.name,
+        url:       data._links.self.href,
+        pipelines: data._embedded.pipelines
```

Comment:
Can we pass the object which holds name and url from here instead of constructing it in Templates.Template